funcx_forwarder/forwarder.py

```
@@ -472,17 +472,41 @@ def handle_results(self):
                 logger.debug(f"Registration message from {message['registration']}")
                 return
 
-            task = RedisTask.from_id(self.redis_pubsub.redis_client, message['task_id'])
+            # only messages with a result or an exception are processed past this point
+            result_or_exception = 'result' in message or 'exception' in message
+            if not result_or_exception:
+                logger.warning("A task result message was received without a result or an exception", extra={
+                    "endpoint_id": endpoint_id,
+                    "endpoint_status_message": message.__dict__
+                })
+                return
+
+            rc = self.redis_pubsub.redis_client
+            task_id = message['task_id']
+
+            if not RedisTask.exists(rc, task_id):
```

Let's add a story to fold this check into `RedisTask.from_id` since that call should raise an exception if the task is missing in Redis.